game/src/edit/roads.rs

```diff
@@ -518,29 +544,25 @@ fn add_new_lane(road: &mut EditRoad, lt: LaneType) -> usize {
             .position(|pair| pair[0].dir != pair[1].dir)
             .map(|x| x + 1)
             .unwrap_or(road.lanes_ltr.len()),
-        // Place on the forwards side, before any sidewalk
+        // Place on the dir side, before any sidewalk
         LaneType::Biking | LaneType::Bus | LaneType::Parking | LaneType::Construction => {
-            if road.lanes_ltr.last().unwrap().lt.is_walkable() {
-                road.lanes_ltr.len() - 1
-            } else {
-                road.lanes_ltr.len()
-            }
+            default_outside_lane_placement(road, dir)
         }
-        // Place it where it's missing
+        // Place it where it's missing, and if both sides have or do not have sidewalks then go with the dir side
```

Comment:
The new road editor doesn't enforce this yet, but sidewalks are only supposed to be at the beginning or end, never in the middle. So I think we can actually keep the previous logic here, since it's a bit simpler. If the road has 0 sidewalks, I don't think it's important to initially place the first one on one side or the other.

I've yet to enforce the "sidewalks only on the outside" rule, because I'm not sure how the "move left/right" control should work for sidewalks. Maybe since they're kind of special, the UI should even just have two checkboxes, "sidewalk on the left" & "sidewalk on the right". Except we do still want to let people set the width... so maybe they're not so special after all. Not sure yet.

Reply:
Agreed, I will simplify it here.